File: src/l4/pkg/l4rust/libl4re-wrapper/lib/scheduler.c
#include <pthread-l4.h>
#include <l4/l4rust/scheduler.h>

EXTERN l4_sched_param_t l4_sched_param_w(unsigned prio,
        l4_cpu_time_t quantum) L4_NOTHROW {
    return l4_sched_param(prio, quantum);
}

EXTERN l4_sched_cpu_set_t l4_sched_cpu_set_w(l4_umword_t offset,
        unsigned char granularity, l4_umword_t map) L4_NOTHROW {
    return l4_sched_cpu_set(offset, granularity, map);
}

EXTERN l4_msgtag_t l4_scheduler_run_thread_w(l4_cap_idx_t scheduler,
        l4_cap_idx_t thread, l4_sched_param_t const *sp) L4_NOTHROW {
    return l4_scheduler_run_thread(scheduler, thread, sp);
}

EXTERN l4_msgtag_t l4_scheduler_info_w(l4_cap_idx_t scheduler,
        l4_umword_t *cpu_max, l4_sched_cpu_set_t *cpus) L4_NOTHROW {
    return l4_scheduler_info(scheduler, cpu_max, cpus);
}

File: src/l4/pkg/l4rust/l4-sys-rust/scheduler.rs
use crate::{l4_sched_param_w, l4_cpu_time_t, l4_umword_t, l4_sched_param_t,
            l4_sched_cpu_set_w, l4_scheduler_run_thread_w, l4_sched_cpu_set_t,
            l4_scheduler_info_w, l4_msgtag_t, l4_cap_idx_t};
use libc::{c_char, c_uint};

#[must_use]
#[inline]
pub fn l4_sched_param(prio: c_uint, quantum: l4_cpu_time_t) -> l4_sched_param_t {
    //Not unsafe because this function is marked as L4_NOTHROW
    unsafe {
        l4_sched_param_w(prio, quantum)
    }
}

#[must_use]
#[inline]
pub fn l4_sched_cpu_set(offset: l4_umword_t, granularity: c_char, map: l4_umword_t) -> l4_sched_cpu_set_t {
    //Not unsafe because this function is marked as L4_NOTHROW
    unsafe {
        l4_sched_cpu_set_w(offset, granularity, map)
    }
}

#[must_use]
#[inline]
pub fn l4_scheduler_run_thread(scheduler: l4_cap_idx_t, thread: l4_cap_idx_t, sp: *const l4_sched_param_t) -> l4_msgtag_t {
    //Not unsafe because this function is marked as L4_NOTHROW
    unsafe {
        l4_scheduler_run_thread_w(scheduler, thread, sp)
    }
}

#[must_use]
#[inline]
pub fn l4_scheduler_info(scheduler: l4_cap_idx_t, cpu_max: &mut l4_umword_t,
        cpus: &mut l4_sched_cpu_set_t) -> l4_msgtag_t {
    //Not unsafe because this function is marked as L4_NOTHROW
    unsafe {
        l4_scheduler_info_w(scheduler, cpu_max, cpus)
    }
}